frontend/src/routes/candidate/(protected)/profile/AvatarSelect.svelte

```diff
@@ -0,0 +1,155 @@
+<script lang="ts">
+  import Field from '$lib/components/common/form/Field.svelte';
+  import Icon from '$lib/components/icon/Icon.svelte';
+  import {onDestroy, onMount} from 'svelte';
+  import {_} from 'svelte-i18n';
+  import {constants} from '$lib/utils/constants';
+  import {authContext} from '$lib/utils/authenticationStore';
+  import {get} from 'svelte/store';
+  import type {Photo} from '$lib/types/candidateAttributes';
+
+  export let photo: Photo | undefined;
+  let photoUrl = photo ? constants.PUBLIC_BACKEND_URL + photo.formats.thumbnail.url : undefined;
+  let imageHasChanged = false;
+  let image: File | undefined;
+
+  export const maxFileSize = 5 * 1024 * 1024; // 5 MB
+
+  let portraitInput: HTMLInputElement | null;
+  let portraitLabel: HTMLLabelElement | null;
+  let portraitImage: HTMLImageElement | null;
+
+  const labelClass = 'w-6/12 label-sm label mx-6 my-2 text-secondary';
+
+  const token = get(authContext.token);
+
+  // function for clicking the portrait input field with space
+  const handlePortraitInput = (event: KeyboardEvent) => {
+    if (event.code === 'Space') {
+      event.preventDefault(); // Prevent default behavior (e.g., scrolling the page)
+      portraitInput?.click();
+    }
+  };
+
+  // add an event listener to the portrait label for keyboard navigation
+  onMount(() => {
+    portraitLabel?.addEventListener('keydown', handlePortraitInput);
+  });
+
+  // remove the event listener on unmount
+  onDestroy(() => {
+    portraitLabel?.removeEventListener('keydown', handlePortraitInput);
+  });
+
+  // change the profile image, does not upload it to strapi
+  const changePhoto = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    if (target.files) {
+      const file = target.files[0];
+      if (file && file.type.startsWith('image/') && file.size <= maxFileSize) {
+        imageHasChanged = true;
+        image = file;
+        let reader = new FileReader();
+
+        reader.onload = (e) => {
+          photoUrl = e.target?.result?.toString();
+        };
+        reader.readAsDataURL(file);
+      }
+    }
+  };
+
+  // upload the current image file to strapi
+  export const uploadPhoto = async () => {
+    if (image && imageHasChanged) {
+      const formData = new FormData();
+      formData.append('files', image);
+
+      const postResponse = await fetch(constants.PUBLIC_BACKEND_URL + '/api/upload/', {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${token}`
+        },
+        body: formData
+      });
+
+      // if photo is already defined and upload was succesful,
+      // delete the old photo
+      if (photo && postResponse.status === 200) {
+        await fetch(constants.PUBLIC_BACKEND_URL + `/api/upload/files/${photo.id}`, {
```

Comment:
Consider using lib/api/candidate.ts where all of the other network request logic (and helper functions for doing so) are